MMDevice/MMDevice.h

```diff
@@ -1428,4 +1445,307 @@ namespace MM {
+       * \param   maxItems Maximum number of items to return
+       * \param   maxItemLength Maximum length of each item name
+       * \return  Status code indicating success or failure
+       */
+      virtual int List(const char* path, char** listOfDatasets, int maxItems, int maxItemLength) = 0;
+
+      /**
+       * \brief   Inserts an image at specific coordinates
+       *
+       * \param   handle Dataset handle
+       * \param   sizeInBytes Size of the image data in bytes
+       * \param   pixels Pointer to the image pixel data
+       * \param   coordinates Array of coordinates for image insertion
+       * \param   numCoordinates Number of coordinate values
+       * \param   imageMeta Image metadata string
+       * \return  Status code indicating success or failure
+       */
+      virtual int AddImage(const char* handle, int sizeInBytes, unsigned char* pixels,
+                           int coordinates[], int numCoordinates, const char* imageMeta) = 0;
+
+      /**
+       * \brief   Appends an image to the dataset
+       * \param   handle Dataset handle
+       * \param   sizeInBytes Size of the image data in bytes
+       * \param   pixels Pointer to the image pixel data
+       * \param   imageMeta Image metadata string
+       * \return  Status code indicating success or failure
+       */
+      virtual int AppendImage(const char* handle, int sizeInBytes, unsigned char* pixels, const char* imageMeta) = 0;
+
+      /**
+       * \brief   Retrieves dataset summary metadata
+       *
+       * \param   handle Dataset handle
+       * \param   [out]  meta Buffer for metadata string allocated by the implementation
+       * \return  Status code indicating success or failure
+       *
+       * \note    Caller must release the metadata buffer using ReleaseStringBuffer()
+       */
+      virtual int GetSummaryMeta(const char* handle, char** meta) = 0;
+
+      /**
+       * \brief   Retrieves metadata for a specific image
+       *
+       * \param   handle Dataset handle
+       * \param   coordinates Array of coordinates identifying the image
+       * \param   numCoordinates Number of coordinate values
+       * \param   [out] meta Buffer for metadata string
+       * \return  Status code indicating success or failure
+       *
+       * \note    Caller must release the metadata buffer using ReleaseStringBuffer()
+       */
+      virtual int GetImageMeta(const char* handle, int coordinates[], int numCoordinates, char** meta) = 0;
+
+      /**
+       * \brief   Retrieves image pixel data
+       * \param   handle Dataset handle
+       * \param   coordinates Array of coordinates identifying the image
+       * \param   numCoordinates Number of coordinate values
+       * \return  Pointer to the image pixel data
+       */
+      virtual const unsigned char* GetImage(const char* handle, int coordinates[],
+         int numCoordinates) = 0;
```

Comment:
Can this have an `int` return value so that it can indicate an error?

Also, how long is the returned pointer valid? If it is not until the handle is closed (which would be problematic), we probably need a "release" function for images (analogous to the one for metadata).